portfolio/src/main/webapp/comments.js

```diff
@@ -36,6 +36,11 @@ class Comments {
         // Reset the commentsListContainer to reload it with the new list of comments.
         this.commentsListContainer.innerHTML = '';
 
+        // If commentsList is empty, don't do anything.
+        if (commentsList.length == 0) {
```

Comment:
Check if commentsList is undefined/null here or you might get an NPE.

Optional: Create a reusable isEmpty function that handles this boolean check.